wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -535,3 +535,35 @@ def icons():
     icon_hooks = hooks.get_hooks('register_icons')
     icons = sorted(itertools.chain.from_iterable(hook([]) for hook in icon_hooks))
     return {'icons': icons}
+
+
+@register.simple_tag
+def format_collection(col: Collection, permitted: QuerySet = None) -> str:
+    """
+    Renders a given Collection's name as a formatted string that displays its
+    hierachrical depth via indenation. If a filtered queryset is supplied,
+    the Collection's depth relative to that queryset is used.
+
+    Example usage: {% format_collection collection collections %}
+    Example output: "&nbsp;&nbsp;&nbsp;&nbsp;&#x21b3 Child Collection"
+    """
+    def _depth(cur_col, count=0):
+        if cur_col.get_parent() in permitted:
+            return _depth(cur_col.get_parent(), count + 1)
```

Comment:
This is going to generate a lot of database queries when looping over a list of collections. Could we perhaps pass a `base_depth` (defaulting to 2) to `format_collection` in place of the `permitted` queryset, so that we only have to do that calculation once per list (and the calculation is presumably just `min(depth)`)?

Reply:
Yeah, I didn't love that algorithm either. I didn't see a better way to do it than what Seb did, but your idea does sound viable. I'll see what I can do.

Reply:
This is fixed in my latest commits.